### PR TITLE
Follows

### PR DESCRIPTION
**Pull request by Ru-T**
## What Changed
- Created follows join table with two user ids - follower and followed
- Associations between follows and users including has_many through
- Created follow and unfollow methods and controller actions, with ability to follow/unfollow other users on their profiles
- Show total followers and following on user profiles
- Create feed with only Tweets by users I'm following
## Why
- To enable users to customize their feed with only certain users

### CODE REVIEW

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -2,4 +2,24 @@
 
 RSpec.describe User, type: :model do
 
-end
\ No newline at end of file
+    describe '#follow' do
+      it "allows a user to follow another" do
+        user1 = FactoryGirl.create(:user)
+        user2 = FactoryGirl.create(:user, email: "newuser2@sb.com")
```
Comment (dkniffin) [inline spec/models/user_spec.rb]: These should be in `let` or `subject` blocks. In most cases, you don't want to do an assignment inside of an `it` block.

File: app/controllers/follows_controller.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+  def create
+    user = User.find(params[:followed_id])
+    current_user.follow(user)
+    redirect_to user, notice: "User has been followed."
+  end
+
+  def destroy
+    user = Follow.find(params[:id]).followed
+    current_user.unfollow(user)
```
Comment (dkniffin) [inline app/controllers/follows_controller.rb:12]: Good job putting the logic in the model here!

File: spec/models/user_spec.rb [hunk 3]
```diff
@@ -2,4 +2,24 @@
...
+    describe '#unfollow' do
+      it "allows a user to unfollow another" do
+        user1 = FactoryGirl.create(:user)
+        user2 = FactoryGirl.create(:user, email: "newuser2@sb.com")
+        user1.follow(user2)
+        user1.unfollow(user2)
+        expect(user1.following?(user2)).to eq false
+        expect(user2.followers).to_not include user1
+      end
+    end
```
Comment (dkniffin) [inline spec/models/user_spec.rb:24]: It looks like you missed `following?`

**Comment by dkniffin** [comment]
Other than the above comments, it looks good :+1:

File: spec/models/user_spec.rb (hunk 3 above)
Comment (Ru-T) [inline spec/models/user_spec.rb:24]: Thank you for the review!!

**Pull request merged**